merlin-worker/src/main/java/gov/nasa/jpl/aerie/merlin/worker/MerlinWorkerAppDriver.java

```diff
@@ -62,10 +62,25 @@ public static void main(String[] args) throws Exception {
       final var planId = new PlanId(notification.planId());
       final var datasetId = notification.datasetId();
 
-      System.out.println(String.format("Received simulation notification for plan Id: %s and data set Id: %s",
-                                       planId,
-                                       datasetId));
+      final Optional<ResultsProtocol.OwnerRole> owner = stores.results().claim(planId, datasetId);
+      if (owner.isEmpty()) continue;
 
+      final var reader = stores.results().lookup(planId);
+      // There is no simulation for this plan
+      if (reader.isEmpty()) continue;
```

Comment:
🔵 Is it okay for there to be no simulation for a plan that does have an entry in `simulation_dataset`? We may be protected from this by some foreign key relationships 🤔 

By "is it okay" I'm wondering if we need to throw some kind of exception... at the same time, it probably doesn't make sense for a random worker to crash... I'm not sure what the best answer is here.

Reply:
I've removed this block of code as it was not needed. It was left over from earlier prototyping efforts.